### ISSUE TITLE
Undefined property: stdClass::$allow_unpublish in paragraphs_bundle_save when upgrading

### ISSUE THREAD

**Issue by markabur** [backdrop-contrib/paragraphs#198]
I'm upgrading from D7 to Backdrop and get a whole bunch of these php notices when I run database updates for the first time.
 
`Undefined property: stdClass::$allow_unpublish in paragraphs_bundle_save`

I've found that adding

```
if (!property_exists($bundle, 'allow_unpublish')) {
  $bundle->allow_unpublish = 1;
}
```
right before `paragraphs_bundle_save($bundle);` in `paragraphs_update_1001()` fixes it.

**Comment by laryn** [comment]
Thanks @markabur -- my only question is whether we should default existing D7 Paragraph bundles to allow unpublishing, or assume they should not be opted in automatically. Any thoughts?

**Comment by markabur** [comment]
Yeah I did think about that. Initially I was leaning toward not opting them in, so that as little as possible changes from Drupal, and to be consistent with this:

```
$form['allow_unpublish'] = array(
  '#type' => 'checkbox',
  '#title' => t('Allow unpublishing from the admin interface.'),
  '#default_value' => (isset($bundle->allow_unpublish)) ? $bundle->allow_unpublish : 0,
);
```


But then I noticed up above that new paragraph bundles get opted in by default, and thought about my Drupal site and how I actually have a "hide this row" field on some bundles for exactly this purpose, and how my editors would probably appreciate that feature on other bundles too. So for me it makes sense to opt-in all the old bundles during the upgrade. But I could definitely see setting it to 0 for the patch.

**Comment by laryn** [comment]
I decided being cautious in the upgrade hook was probably best. Look okay @markabur? https://github.com/backdrop-contrib/paragraphs/pull/202

**Comment by laryn** [comment]
I just found/merged your PR with the tweak discussed; it wasn't linked here in the way I'm used to and I missed it. (If the first comment follows this exact pattern, it gets more deeply connected to the issue:   `Fixes #198.`)
